ruma-identity-service-api/src/keys/get_public_key/v2.rs

```diff
@@ -0,0 +1,40 @@
+//! [GET /_matrix/identity/v2/pubkey/{keyId}](https://matrix.org/docs/spec/identity_service/r0.3.0#get-matrix-identity-v2-pubkey-keyid)
+
+use ruma_api::ruma_api;
+use ruma_identifiers::KeyId;
+
+ruma_api! {
+    metadata: {
+        description: "Get the public key for the passed key ID.",
+        method: GET,
+        name: "get_public_key",
+        path: "/_matrix/identity/v2/pubkey/:key_id",
+        rate_limited: false,
+        authentication: None,
+    }
+
+    request: {
+        /// The ID of the key.
+        #[ruma_api(path)]
+        pub key_id: KeyId<String, String>
+    }
+
+    response: {
+        /// Unpadded Base64 encoded public key.
+        pub public_key: String
+    }
+}
+
+impl Request {
+    /// Create a `Request` with the given key_id.
+    pub fn new<A, K>(key_id: KeyId<String, String>) -> Self {
```

Comment:
I'm surprised this compiles without warnings. This function should not be generic.

Reply:
Oh my bad, forgot the remove those, sorry.

Reply:
Removed the useless generics.